internal/host/conditions.go

```diff
@@ -92,3 +94,11 @@ func (v *validator) isSuccessfulContainerImageAvailability(c *validationContext)
 	imagesStatuses, err := common.UnmarshalImageStatuses(c.host.ImagesStatus)
 	return err == nil && len(imagesStatuses) > 0 && allImagesValid(imagesStatuses)
 }
+
+func (v *validator) isHostStageTimedOut(c *validationContext) bool {
+	return c.infraEnv == nil && c.host != nil && c.host.Progress != nil && c.host.Progress.StageTimedOut
```

Comment:
how does infra env related here?

Reply:
If infra env is not nil then the host is unbound

Reply:
But maybe it can be removed